src/vehicle.h

```diff
@@ -112,6 +112,8 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
 
     vehicle_part( const vpart_str_id& vp, int dx, int dy, item&& it );
 
+    explicit operator bool() const;
```

Comment:
I've added documentation
```
Check this instance is non-null (not default constructed)
```

Useful when checking the response from `veh_interact::select_part` and is consistent with `item_location`, `string_id` etc.